From 0f9f3d387ce24c4f70bacbbe545fd4d3c76a9274 Mon Sep 17 00:00:00 2001
From: Eric Leblond <eric@regit.org>
Date: Fri, 4 Mar 2016 10:53:21 +0100
Subject: [PATCH 1/2] output-json: add ethernet headers

This patch adds logging of ethernet headers in the JSON output.
---
 src/output-json.c | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/src/output-json.c b/src/output-json.c
index 32495d41..78d7a69d 100644
--- a/src/output-json.c
+++ b/src/output-json.c
@@ -470,6 +470,41 @@ json_t *CreateJSONHeader(const Packet *p, int direction_sensitive,
             break;
     }
 
+    if (p->ethh) {
+        json_t *ejs = json_object();
+        if (ejs != NULL) {
+            char eth_addr[19];
+            uint8_t *src, *dst;
+
+            if ((PKT_IS_TOSERVER(p))) {
+                src = p->ethh->eth_dst;
+                dst = p->ethh->eth_src;
+            } else {
+                src = p->ethh->eth_src;
+                dst = p->ethh->eth_dst;
+            }
+
+            json_object_set_new(ejs, "type", json_integer(ntohs(p->ethh->eth_type)));
+            snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                    src[0],
+                    src[1],
+                    src[2],
+                    src[3],
+                    src[4],
+                    src[5]);
+            json_object_set_new(ejs, "src", json_string(eth_addr));
+            snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                    dst[0],
+                    dst[1],
+                    dst[2],
+                    dst[3],
+                    dst[4],
+                    dst[5]);
+            json_object_set_new(ejs, "dst", json_string(eth_addr));
+            json_object_set_new(js, "ether", ejs);
+        }
+    }
+
     return js;
 }
 
-- 
2.11.0

